strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/io/ArtifactOutputStream.java

```diff
@@ -15,18 +20,118 @@
 public class ArtifactOutputStream
         extends MultipleDigestOutputStream
 {
-
+    private Function<byte[], String> digestStringifier = MessageDigestUtils::convertToHexadecimalString;
     private ArtifactCoordinates coordinates;
+    /**
+     * Used to cache Artifact contents if needed.
+     */
+    private OutputStream cacheOutputStream;
```

Comment:
We don't really want to cache artifacts in memory. What is this relating to? Is it connected to the `ReloadableInputStreamHandler`?